nvflare/job_config/base_fed_job.py

```diff
@@ -0,0 +1,165 @@
+from nvflare.app_common.abstract.model_locator import ModelLocator
+from nvflare.app_common.abstract.model_persistor import ModelPersistor
+from nvflare.app_common.widgets.convert_to_fed_event import ConvertToFedEvent
+from nvflare.app_common.widgets.intime_model_selector import IntimeModelSelector
+from nvflare.app_common.widgets.streaming import AnalyticsReceiver
+from nvflare.app_common.widgets.validation_json_generator import ValidationJsonGenerator
+from nvflare.app_opt.tracking.tb.tb_receiver import TBAnalyticsReceiver
+from nvflare.job_config.api import FedJob, validate_object_for_job
+from nvflare.job_config.script_runner import FrameworkType
+
+
+class BaseFedJob(FedJob):
+    """Unified BaseFedJob that supports PyTorch, TensorFlow, and Scikit-learn frameworks.
+
+    This class consolidates the previously separate PT and TF BaseFedJob implementations
+    into a single unified interface that can handle all frameworks.
+    """
+
+    def __init__(
+        self,
+        initial_model: Any = None,
+        initial_params: Optional[Dict] = None,
+        name: str = "fed_job",
+        min_clients: int = 1,
+        mandatory_clients: Optional[List[str]] = None,
+        key_metric: str = "accuracy",
+        validation_json_generator: Optional[ValidationJsonGenerator] = None,
+        intime_model_selector: Optional[IntimeModelSelector] = None,
+        convert_to_fed_event: Optional[ConvertToFedEvent] = None,
+        analytics_receiver: Optional[AnalyticsReceiver] = None,
+        model_persistor: Optional[ModelPersistor] = None,
+        model_locator: Optional[ModelLocator] = None,
+        framework: FrameworkType = FrameworkType.PYTORCH,
+    ):
+        """Unified BaseFedJob for PyTorch, TensorFlow, and Scikit-learn.
+
+        Configures ValidationJsonGenerator, IntimeModelSelector, AnalyticsReceiver, ConvertToFedEvent.
+
+        User must add controllers and executors.
+
+        Args:
+            initial_model: Initial model object. Can be:
+                - nn.Module for PyTorch
+                - tf.keras.Model for TensorFlow
+                - None if using initial_params for sklearn
+            initial_params: Initial model parameters (dict). Used for sklearn.
+                If provided, initial_model should be None.
+            name: Name of the job. Defaults to "fed_job".
+            min_clients: The minimum number of clients for the job. Defaults to 1.
+            mandatory_clients: Mandatory clients to run the job. Default None.
+            key_metric: Metric used to determine if the model is globally best.
+                If metrics are a dict, key_metric can select the metric used for global model selection.
+                Defaults to "accuracy".
+            validation_json_generator: A component for generating validation results.
+                If not provided, a ValidationJsonGenerator will be configured.
+            intime_model_selector: A component for selecting the model.
```

Comment:
model_selector,
could you try enhance the docstring as well?
when is this model_selector being called?